src/compile/common.ts

```diff
@@ -101,17 +106,16 @@ export function formatSignalRef(
       )
     };
   } else {
-    return {
-      signal: `''+${vgField(fieldDef, {expr})}`
-    };
+    return {signal: `''+${vgField(fieldDef, {expr})}`};
   }
 }
 
 /**
  * Returns number format for a fieldDef
  */
 export function numberFormat(fieldDef: TypedFieldDef<string>, specifiedFormat: string, config: Config) {
-  if (isTimeFieldDef(fieldDef)) {
+  const formatType = getFormatType(fieldDef);
+  if (formatType === 'time' || (!formatType && isTimeFieldDef(fieldDef))) {
```

Comment:
It feel wrong that a method named `numberFormat` can return a time format.  Perhaps, a bit refactor needs to be done here.